### PR TITLE
Add yourVaccine to Calculator core

### PR DESCRIPTION
**Pull request by beshaya**
(no ux change)

### CODE REVIEW

**Comment by blanchardjeremy** [comment]
@beshaya  I assume you wanna fix the typecheck before I do a review.  Is that right?

**Comment by beshaya** [comment]
> @beshaya I assume you wanna fix the typecheck before I do a review. Is that right?

whoops, fixed!

File: src/data/__tests__/calculate.test.ts [hunk 1]
```diff
@@ -432,9 +426,82 @@ describe('calculate', () => {
...
+      }
+      expect(calcValue(vaccineLongScenario)).toBeCloseTo(
+        calcValue(noVaccineLongScenario)! * 0.2,
+      )
+    })
+
+    it('Should apply vaccine multiplier to housemates activities', () => {
+      const noVaccineHousemate = {
+        ...noVaccineScenario,
+        interraction: 'repeated',
```
Comment (jy2wong) [inline src/data/__tests__/calculate.test.ts:479]: interaction

File: src/locales/en.json [hunk 2]
```diff
@@ -216,6 +216,12 @@
       "microcovid_budget_one_percent": "Keeps to a budget of 200 microCOVID per week (\"Standard Caution Budget\")",
       "microcovid_budget_deci_percent": "Keeps to a budget of 20 microCOVID per week (\"High Caution Budget\")",
       "hasCovid": "Has COVID"
+    },
+    "vaccine": {
+      "pfizer": "Pfizer BioNTech",
+      "moderna": "Moderna",
+      "astra_zeneca": "AstraZeneca",
+      "unkown": "Unknown or other"
```
Comment (jy2wong) [inline src/locales/en.json:224]: unkown -> unknown

**Review by jy2wong — CHANGES_REQUESTED** [review]

File: src/data/__tests__/calculate.test.ts (hunk 1 above)
Comment (beshaya) [reply src/data/__tests__/calculate.test.ts]: done

File: src/locales/en.json (hunk 2 above)
Comment (beshaya) [reply src/locales/en.json]: done

File: src/data/calculate.ts [hunk 3]
```diff
@@ -231,14 +248,34 @@ export const calculatePersonRiskEach = (
   }
 }
 
+const getVaccineMultiplier = (data: CalculatorData): number | null => {
+  if (data.yourVaccineType === '') {
+    return 1
+  }
+  const vaccineMultipliers = Vaccines[data.yourVaccineType].multipliers
+  if (
+    data.yourVaccineDoses >= vaccineMultipliers.length ||
```
Comment (blanchardjeremy) [inline src/data/calculate.ts:257]: Took me a while to understand this. I think the name "multipliers" is vague in this case. I'd suggest something that more clearly says "here is a list of multiplier for this vaccine based on the number of doses they've received." I don't have a particular suggestion. Maybe multipliersByDose or something. I'd definitely chang it here. And I'd consider changing it in data.ts and any place that it's referenced.

File: src/data/calculate.ts [hunk 4]
```diff
@@ -231,14 +248,34 @@ export const calculatePersonRiskEach = (
...
+
 export const calculateActivityRisk = (data: CalculatorData): number | null => {
   try {
     if (data.interaction === '') {
       return null
     }
 
+    const vaccineMultiplier = getVaccineMultiplier(data)
+    if (vaccineMultiplier === null) {
+      console.log(data.yourVaccineType + ' ' + data.yourVaccineDoses)
```
Comment (blanchardjeremy) [inline src/data/calculate.ts:273]: I assume you don't want this in anymore? Or do you want it removed once we finish the vaccine stuff? If so, we wanna add a noticeable comment next to it.

File: src/data/calculate.ts [hunk 5]
```diff
@@ -276,7 +313,8 @@ export const calculateActivityRisk = (data: CalculatorData): number | null => {
     if (multiplier > MAX_ACTIVITY_RISK) {
       multiplier = MAX_ACTIVITY_RISK
     }
-    return multiplier
+
+    return multiplier * vaccineMultiplier
```
Comment (blanchardjeremy) [inline src/data/calculate.ts:317]: God. This is just such a satisfying line to read. It's like.... go humanity! We did it! We have vaccines!

**Review by blanchardjeremy — APPROVED** [review]
LGTM after the minor changes below are addressed.

File: src/data/calculate.ts (hunk 5 above)
Comment (beshaya) [reply src/data/calculate.ts:317]: 👩‍⚕️💉😎

File: src/data/calculate.ts (hunk 3 above)
Comment (beshaya) [reply src/data/calculate.ts]: Done.

File: src/data/calculate.ts (hunk 4 above)
Comment (beshaya) [reply src/data/calculate.ts]: Removed.

**Pull request merged**